services/offline/src/lib/online-status.ts

```diff
@@ -28,21 +28,16 @@ export function useOnlineStatus(options?: OnlineStatusOptions): OnlineStatus {
     // initialize state to `navigator.onLine` value
     const [online, setOnline] = useState(navigator.onLine)
 
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    const updateState = useCallback(
-        debounce(
-            ({ type }: Event) => setOnline(type === 'online'),
-            options?.debounceDelay || 1000
-        ),
-        [options]
```

Comment:
Simply changing this to `options?.debounceDelay` would solve the cancellation issue without adding an unnecessary dependency - the issue is that `options` might be referentially instable if created during the parent render loop.  However, the integer `debounceDelay` should be stable even across re-renders so long as the value doesn't change.  I would prefer updating this dependency rather than adding a dependency just for debouncing.

Reply:
(note that `app-runtime` is explicitly conservative in externaly dependencies - it currently has none, though `react-query` will be added soon

Reply:
> (note that app-runtime is explicitly conservative in adding external dependencies - it currently has none, though react-query will be added soon)

`lodash` wasn't explicitly defined in `package.json` :grimacing: 

Would you prefer we add https://github.com/xnimorz/use-debounce/blob/master/src/useDebouncedCallback.ts (or https://github.com/lodash/lodash/blob/master/debounce.js) under `services/offline/src/lib`?

Reply:
I implemented this suggestion - is it okay to add lodash to deps?